osbrain/agent.py

```diff
@@ -930,11 +933,59 @@ def _subscribe(self, alias: str, handlers: Dict[Union[bytes, str], Any]):
             handlers = {'': handlers}
 
         curated_handlers = topics_to_bytes(handlers)
+
         # Subscribe to topics
-        for topic in curated_handlers.keys():
-            self.socket[alias].setsockopt(zmq.SUBSCRIBE, topic)
+        for topic in curated_handlers:
+            self._subscribe_to_topic(alias, topic)
+
         # Reset handlers
-        self._set_handler(self.socket[alias], curated_handlers)
+        if isinstance(self.address[alias], AgentChannel):
+            channel = self.address[alias]
+            sub_address = channel.receiver
+            uuid = channel.uuid
+            curated_handlers = topics_to_bytes(handlers, uuid=uuid)
+            self._set_handler(self.socket[sub_address], curated_handlers)
+        else:
+            self._set_handler(self.socket[alias], curated_handlers)
```

Comment:
Maybe this is why it fails. It seems we are settings `curated_handlers`, but what you actually want to do is to update this dictionary if it existed before? (not sure, did not run the tests)